db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -1,38 +1,221 @@
 package edu.artic.db
 
+import android.util.Log
+import com.fuzz.rx.asObservable
+import com.fuzz.rx.disposedBy
+import edu.artic.db.models.ArticAppData
+import edu.artic.db.models.ArticEvent
+import edu.artic.db.models.ArticExhibition
 import io.reactivex.Observable
-import org.threeten.bp.LocalDateTime
-import org.threeten.bp.format.DateTimeFormatterBuilder
+import io.reactivex.functions.BiFunction
 import javax.inject.Inject
 
-class AppDataManager @Inject constructor(private val serviceProvider: AppDataServiceProvider) {
+class AppDataManager @Inject constructor(
+        private val serviceProvider: AppDataServiceProvider,
+        private val appDataPreferencesManager: AppDataPreferencesManager,
+        private val appDatabase: AppDatabase
+) {
+    companion object {
+        const val HEADER_LAST_MODIFIED = "last-modified"
+    }
+
+    fun loadData(): Observable<ProgressDataState> {
 
-    fun getBlob(): Observable<AppDataState> {
+        //First load app data. if empty is returned, do nothing, otherwise download everything else.
+        return Observable.create<ProgressDataState> { observer ->
+            getBlob().subscribe({
+                when (it) {
+
+                    is ProgressDataState.Done<*> -> {
+                        loadPostAppData()
+                                .subscribe({ amountDownload ->
+                                    Log.d("AppDataManager", "Amount downloaded $amountDownload")
```

Comment:
this exact set of code is duplicated below. Make a method consolidating it.

Reply:
it's same as `Empty`.

Reply:
already changed, guess I didn't push it up. thanks